sycl/source/detail/scheduler/commands.cpp

```diff
@@ -2118,6 +2118,39 @@ cl_int enqueueImpKernel(
   return PI_SUCCESS;
 }
 
+cl_uint enqueueReadWriteHostPipe(const QueueImplPtr &Queue,
+                                 const std::string &KernelName,
+                                 const std::string &PipeName, bool blocking,
+                                 void *ptr, size_t size,
+                                 std::vector<RT::PiEvent> &RawEvents,
+                                 RT::PiEvent *OutEvent, bool read) {
+  detail::OSModuleHandle M =
+      detail::OSUtil::getOSModuleHandle("HostPipeReadWriteKernelName");
+  RT::PiProgram Program =
+      sycl::detail::ProgramManager::getInstance().getBuiltPIProgram(
+          M, Queue->getContextImplPtr(), Queue->getDeviceImplPtr(), KernelName);
+
```

Comment:
Only need get the program base on the pipe type, require frontend to provide the info.